### PR TITLE
Decouple metrics retrieval from vertx core + cleanup

### PR DESCRIPTION
**Pull request by vietj**
@purplefox please review, thanks

### CODE REVIEW

**Comment by purplefox** [comment]
Can you rebase this with master? Has conflicts

**Comment by vietj** [comment]
rebased

**Comment by vietj** [comment]
is it ok to merge ?

File: src/main/java/examples/MetricsExamples.java [hunk 1]
```diff
@@ -30,10 +30,12 @@
 public class MetricsExamples {
```
Comment (purplefox) [inline src/main/java/examples/MetricsExamples.java]: I guess this should be deleted and any metrics examples should be added to the vertx-examples repo instead

**Comment by purplefox** [comment]
Looks good to me.

**Comment by vietj** [comment]
thanks, I'll migrate relevant doc + examples to vertx-metrics soon.

File: src/main/java/examples/MetricsExamples.java (hunk 1 above)
Comment (vietj) [inline src/main/java/examples/MetricsExamples.java]: yes I wanted to validate first the core changes and refactoring, will do soon with cleanup and publish docs in stack page

**Pull request merged**